core/src/main/java/org/elasticsearch/ingest/PipelineConfiguration.java

```diff
@@ -46,7 +50,9 @@
             XContentBuilder contentBuilder = XContentBuilder.builder(parser.contentType().xContent());
             XContentHelper.copyCurrentStructure(contentBuilder.generator(), parser);
             builder.setConfig(contentBuilder.bytes());
+            builder.setXContentType(contentBuilder.contentType());
```

Comment:
can we rather pass it in with the bytes? I think it would be better than two setters